tests-integration/src/tests/instance_test.rs

```diff
@@ -151,6 +191,75 @@ PARTITION BY RANGE COLUMNS (ts) (
     check_output_stream(output, expected).await;
 }
 
+#[apply(both_instances_cases)]
+async fn test_show_create_external_table(instance: Arc<dyn MockInstance>) {
+    let fe_instance = instance.frontend();
+    let format = "csv";
+    let location = find_testing_resource("/tests/data/csv/");
+    let table_name = "various_type_csv";
+
+    let output = execute_sql(
+        &fe_instance,
+        &format!(
+            r#"create external table {table_name} with (location='{location}', format='{format}', pattern='various*');"#,
+        ),
+    )
+    .await;
+    assert!(matches!(output, Output::AffectedRows(0)));
+
+    let output = execute_sql(&fe_instance, &format!("show create table {table_name};")).await;
+
+    let expect = if instance.is_distributed_mode() {
+        format!(
+            r#"+------------------+----------------------------------------------------------------------+
+| Table            | Create Table                                                         |
++------------------+----------------------------------------------------------------------+
+| various_type_csv | CREATE EXTERNAL TABLE IF NOT EXISTS "various_type_csv" (             |
+|                  |   "c_int" BIGINT NULL,                                               |
+|                  |   "c_float" DOUBLE NULL,                                             |
+|                  |   "c_string" DOUBLE NULL,                                            |
+|                  |   "c_bool" BOOLEAN NULL,                                             |
+|                  |   "c_date" DATE NULL,                                                |
+|                  |   "c_datetime" TIMESTAMP(0) NULL,                                    |
+|                  |                                                                      |
+|                  | )                                                                    |
+|                  |                                                                      |
+|                  | ENGINE=file                                                          |
+|                  | WITH(                                                                |
+|                  |   format = 'csv',                                                    |
+|                  |   location = '{location}', |
```

Comment:
I found that this is a flaky test because it is depending on the current dir name length....

Reply:
For sqlness tests, the output's format can be changed by some sqlness hints (the lines started with "--"). See the examples in "join_10_tables.result". 

For rust tests, you can simply strip the create table SQL out of the output recordbatch's vectors, and parse the SQL again to get the table options for comparation.